added the instruction for how the game will work

YoseptF opened the pull request:
We have added all the skeleton for how the game will work.
We added some comments for readability, and 2 extra puts to avoid Rubocop marking black if's as incorrect
![image](https://user-images.githubusercontent.com/44252641/71531338-e1286780-28b3-11ea-9d6a-ba54f660d67c.png)

YoseptF committed:
added the instruction for how the game will work

YoseptF committed:
spacing

shshamim63 commented:
## Status :registered:  [!! Required Changes !!] :registered: 
___
Hi @YoseptF and @lucasmazo32 :wave: ,
The PR contains a meaningful title and definitely well organized summary about the task that is helpful to get the overview about this milestone. Remember that you will have to enclose all of the UI related print/puts in the main file ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/main_rb.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). Based on that there are some output that are missing in here.
___
#### Please follow the instruction below to complete this milestone.
___
###  :spaghetti: [Mandatory Changes] :spaghetti: 
___
1. From the condition of the while it seems that it will perform some extra loop. Draw will check if all the place of the board is taken. And iteration of upto 9 also indicates the same thing. Well only five move is enough to terminate a game with a winning result. 
__________
_X_|_X_|_X_
_O_|_O_|__
___|___|___
https://github.com/lucasmazo32/tic-tac-toe/blob/458255ad593f114a21e2032ab023adf5e3fd73fe/bin/main.rb#L24-L33
I think the loop should be something 
```
def game_end
   draw || winner
end
until game_end
# all the business logic
end
```
2. Game tells which player turn it is. There should be a message to notify user about their turn.
3. Game asks player to select from available moves. If a place is already taken it should render that the place is already taken or input is invalid if no such cell exist in the board.
___
### :1234: [Optional Changes]
1. May be you can ask for players name.
___
### Please fix the above mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](__Shakhawat Hossain__)

YoseptF committed:
changes after first review, and some explanations

YoseptF committed:
rubocop changes

YoseptF commented:
- [ ] We think both conditions are virtually the same `while a && b` and `until a || b` are both waiting for one of them to be false, maybe the confusion comes from the initial state of our condition
![image](https://user-images.githubusercontent.com/44252641/71533423-2f426880-28be-11ea-918b-e422d7da30c1.png)
while draw == true `AND` turns < 9 the game continue looping, this is because we are assuming a draw from the beginning and changing it to false in case someone wins (the game also ends if 9 turns have passed)
- [ ] We had a dummy text in place (It said player mechanics here, talking with the player was intended to be inside this block, but we added some puts to make it clearer)
![image](https://user-images.githubusercontent.com/44252641/71533499-a7109300-28be-11ea-9260-3fd090a5f806.png)
- [ ] We added a method to check the availability of space, and also to check the number and the size of said number.
![image](https://user-images.githubusercontent.com/44252641/71533537-e63ee400-28be-11ea-9e36-dfbc50338cc8.png)

- [ ] Finally, we are working on how to translate all of this to objects in the next milestone :+1:

shshamim63 commented:
## status :green_apple: [Approved]
___
Hi @YoseptF and @lucasmazo32 :wave,
It is me again. Thanks for clarifying your point of view. Now why I was talking about the loop, cause, you have predefined that initially the game will be a draw. But naturally that should get decided based on the players moves. Now the mechanism part. A set of actions will execute based on users turn. So I think it is better to gather all the specific individual things at the initial mode.
___
### This :+1:  is from me. You can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](__Shakhawat Hossain__)
